File: test/counter/counter_test.py
import sys

sys.path.extend([".", "../.."])
import python as specimen

counter = [0]


@specimen.run(
    specimen.read_local_file("counter_data.yaml", location=__file__),
)
def test(context, expected_count=None, **kwargs):
    try:
        if expected_count is not None:
            context.test_case.assertEqual(
                counter[0], expected_count, "count comparison"
            )
    finally:
        counter[0] += 1

File: rust/tests/counter_test.rs
#[test]
fn test_counter() {
    let mut counter = 0;

    specimen::run(
        &mut |tile: &specimen::Dict| -> Result<(), Box<str>> {
            match tile.get("expected_count") {
                Some(expected_count) => {
                    let expected_count = expected_count.parse().unwrap();
                    if counter != expected_count {
                        return Err(Box::from(format!(
                            "Counter ({counter}) did not match expected count ({expected_count})",
                        )));
                    }
                }
                None => {}
            };
            counter += 1;
            Ok(())
        },
        &[specimen::file::File::read_local_file(
            "../test/counter/counter_data.yaml",
        )],
    )
}
